test/unit/gateways/paymill_test.rb

```diff
@@ -181,6 +181,32 @@ def test_successful_store
     assert response.test?
   end
 
+  def test_store_includes_currency
+    expected_currency = 'USD'
+
+    @gateway.expects(:raw_ssl_request).with(
```

Comment:
Looking through the other unit tests in this file, none of them use `.with`. Why is it needed for this unit test?

Reply:
The purpose of the test is to ensure that the currency gets included in the request sent to Paymill. So, the expectation is that when we call the method to make an SSL request, it's done ```with``` the currency parameter set to the right value.

Reply:
Okay, that makes sense. Why was this check added as a unit test but not as a remote test?

Reply:
Good question. When Paymill responds to a ```store``` request, it doesn't return anything different in the case of a currency and amount being sent vs. not being sent. So, there's nothing that we can check in a remote test. The options are being set in the code that implements the request, but a user of that code cannot see those details. The remote test is performing its testing as if it were a user of the code, only calling the public methods (e.g., ```gateway.store```) and checking the return values. The more general theme here is unit testing vs. integration testing.

Reply:
I remember that there was a similar issue when trying to test the new `tx_source` GSF for the MiGS gateway. Would replicating the approach I took for those remote make sense here?
https://github.com/activemerchant/active_merchant/commit/4493b55b5bf49f854bcd43e08cae1b9e30b62eb2#diff-a1a05cf6a8356f8c6539f34382a90bf6R110

Reply:
That is indeed a similar case. I don't think a test like that is possible in this case, though, because the options being passed are hardcoded in the ```store``` implementation. There isn't a way for the caller (i.e. our remote test) to change them to something invalid. FWIW, I did manually test that.